__tests__/endpoint-requests.test.js

```diff
@@ -63,23 +63,116 @@ describe("GET /api/articles", () => {
+      });
+  });
+  test('status: 200, should sort articles by date by default ', () => {
+    return request(app)
+      .get('/api/articles')
+      .expect(200)
+      .then((response) => {
+        const articles = response.body.articles;
+        expect(articles).toBeSortedBy(articles.created_at, {
+          descending: true,
+        });
+      });
+  });
+  test('status: 200, should accept a query: sort_by that allows a user to sort articles by any valid column ', () => {
+    return request(app)
+      .get('/api/articles?sort_by=author')
+      .expect(200)
+      .then((response) => {
+        const articles = response.body.articles;
+        expect(articles).toBeSortedBy(articles.author, {
+          descending: true,
+        });
+      });
+  });
+  test('status: 200, should sort queries by descending by default', () => {
+    return request(app)
+      .get('/api/articles?sort_by=author')
+      .expect(200)
+      .then((response) => {
+        const articles = response.body.articles;
+        expect(articles).toBeSortedBy(articles.author, {
+          descending: true,
+        });
+      });
+  });
+  test('status: 200, should accept a query order that allows the user change the sort order', () => {
+    return request(app)
+      .get('/api/articles?sort_by=author&order=asc')
+      .expect(200)
+      .then((response) => {
+        const articles = response.body.articles;
+        expect(articles).toBeSortedBy(articles.author);
+      });
+  });
+
+  test('status: 200, should allow for mixed case queries', () => {
+    return request(app)
+      .get('/api/articles?toPic=miTCh&soRt_by=AuTHor&order=Asc')
+      .expect(200)
+      .then((response) => {
+        const articles = response.body.articles;
+        expect(articles).toHaveLength(11);
+        expect(articles).toBeSortedBy(articles.author);
+      });
+  });
+
+  test('status: 400, should return bad request when user tries to filter by an invalid topic ', () => {
+    return request(app)
+      .get('/api/articles?topic=hats')
+      .expect(400)
+      .then((response) => {
+        const error = response.body.msg;
+        expect(error).toBe('bad request - invalid topic');
+      });
+  });
+  test('status: 400, should return bad request when a user tries to sort_by an invalid column', () => {
+    return request(app)
+      .get('/api/articles?topic=cats&sort_by=goiejkioe')
+      .expect(400)
+      .then((response) => {
+        const error = response.body.msg;
+        expect(error).toBe('bad request - cannot sort results by that column');
+      });
+  });
+  test('status: 400, should return bad request when a user tries to order the results by an invalid value', () => {
+    return request(app)
+      .get('/api/articles?topic=cats&sort_by=author&order=up')
+      .expect(400)
+      .then((response) => {
+        const error = response.body.msg;
+        expect(error).toBe('bad request - invalid sort order');
+      });
+  });
+});
+
```

Comment:
really great testing. have a think about testing for if the query is valid but there are no results